### PR TITLE
Cfr variant analysis

### PR DESCRIPTION
**Pull request by seabbs**
Includes a model of deaths by date of infections ~ cases by date of infection modulated by covariates and the proportion of the novel variant. Evaluates both an additive and a multiplicative effect.  Best fitting model adjusts for UTLA, region, and cases (though a better fit is achieved by including a spline on time either nationally or regionally). It is likely that these results are confounded by the age distribution of cases which typically varies over time and space. 

Sample (do not interpret the effect sizes below) output (from multiplicative model): 

```
 Family: variant_nb 
  Links: mu = logit; phi = identity; alpha = identity 
Formula: deaths ~ s(normalised_cases, k = 5) + region + (1 | utla) 
   Data: data (Number of observations: 1503) 
Samples: 4 chains, each with iter = 2000; warmup = 1000; thin = 1;
         total post-warmup samples = 4000

Smooth Terms: 
                         Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
sds(snormalised_cases_1)     4.39      1.73     2.15     8.75 1.00     1395     1998

Group-Level Effects: 
~utla (Number of levels: 148) 
              Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
sd(Intercept)     0.55      0.04     0.47     0.63 1.00      594     1324

Population-Level Effects: 
                            Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
Intercept                      -4.22      0.11    -4.43    -4.01 1.01      438     1060
regionLondon                   -1.20      0.15    -1.49    -0.90 1.01      473      882
regionMidlands                  0.38      0.12     0.14     0.62 1.01      586     1479
regionNorthEastandYorkshire     0.46      0.13     0.21     0.72 1.01      427      800
regionNorthWest                 0.51      0.13     0.25     0.77 1.01      522     1073
regionSouthEast                -0.29      0.13    -0.53    -0.03 1.00      671     1415
regionSouthWest                -0.18      0.16    -0.49     0.13 1.01      550     1241
snormalised_cases_1            -8.20      0.98   -10.13    -6.28 1.00     2640     2830

Family Specific Parameters: 
      Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
phi       8.72      0.65     7.53    10.05 1.00     2807     2694
alpha     3.15      0.20     2.77     3.55 1.00     2890     2894

Samples were drawn using sampling(NUTS). For each parameter, Bulk_ESS
and Tail_ESS are effective sample size measures, and Rhat is the potential
scale reduction factor on split chains (at convergence, Rhat = 1).
```

Running all results overnight. Not sure this should go anywhere. Method itself seems fine.  An additional sensitivity analysis to do here is use the Rts from death data in the original model where the effect should in theory be the same if the   CFR and age distribution are the same for both strains.

### CODE REVIEW

**Comment by sbfnk** [comment]
What do the results look like?

**Comment by seabbs** [comment]
```
$all
                                          elpd_diff se_diff
all_with_regional_residuals_additive          0.0       0.0
all_with_regional_residuals_multiplictive   -37.6      12.1
all_with_residuals_additive                 -64.0      13.2
all_with_residuals_multiplictive            -82.5      15.4
all_additive                               -426.2      32.1
all_multiplictive                          -440.2      28.6
utla_additive                              -605.6      32.5
utla_multiplictive                         -616.9      31.2
time_multiplictive                         -804.4      38.3
time_additive                              -809.8      38.2
region_multiplictive                       -839.4      36.0
region_additive                            -842.2      36.3
cases_multiplictive                       -1060.9      39.6
intercept_additive                        -1060.9      39.6
intercept_multiplictive                   -1060.9      39.6
cases_additive                            -1061.0      39.5
```

```
$multiplicative
$multiplicative$intercept
 2.5%   50% 97.5% 
-0.51 -0.37 -0.22 

$multiplicative$cases
 2.5%   50% 97.5% 
-0.50 -0.36 -0.20 

$multiplicative$region
 2.5%   50% 97.5% 
  0.8   1.2   1.6 

$multiplicative$time
 2.5%   50% 97.5% 
-0.83 -0.77 -0.70 

$multiplicative$utla
 2.5%   50% 97.5% 
 0.78  1.10  1.50 

$multiplicative$all
 2.5%   50% 97.5% 
  1.8   2.1   2.6 

$multiplicative$all_with_residuals
 2.5%   50% 97.5% 
-0.51 -0.37 -0.21 

$multiplicative$all_with_regional_residuals
 2.5%   50% 97.5% 
-0.75 -0.63 -0.49 


$additive
$additive$intercept
   2.5%     50%   97.5% 
-0.0120 -0.0083 -0.0046 

$additive$cases
   2.5%     50%   97.5% 
-0.0110 -0.0076 -0.0039 

$additive$region
  2.5%    50%  97.5% 
0.0084 0.0120 0.0150 

$additive$time
  2.5%    50%  97.5% 
-0.026 -0.023 -0.020 

$additive$utla
  2.5%    50%  97.5% 
0.0056 0.0075 0.0096 

$additive$all
  2.5%    50%  97.5% 
0.0094 0.0110 0.0130 

$additive$all_with_residuals
  2.5%    50%  97.5% 
0.0021 0.0034 0.0046 

$additive$all_with_regional_residuals
  2.5%    50%  97.5% 
0.0037 0.0050 0.0063 
```

Not keen on the difference in direction between the two models when you adjust for time.

File: R/cfr.r [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+         time = (time - mean(time)) / sd(time))
+
+# Define model ------------------------------------------------------------
+# D = c^{+}(1-f)C + c^{-}fC
+# where c is the CFR, f is the fraction of cases that are sgtf, C is the number
+# of cases by date of infection, and D is the number of deaths by date of infection
+# effect of the variant on the CFR is assumed to be:
+# c^{-} = \alpha c^{+} or c^{-} = \alpha + c^{+}
+# leading to:
+# D = (1 - (\alpha - 1)f)c^{+}C or D = (c^{+} + \alpha f)C
```
Comment (sbfnk) [inline R/cfr.r:70]: should be D = (1 + (\alpha - 1)f)c^{+}C
(i.e., I think it's correct in the function below)

File: R/cfr.r [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+warning("Fitting models sequentially due to mclapply issues")
+fits <- list()
+fits[["multiplicative"]] <- lapply(models, nb_model)
+fits[["additive"]] <- lapply(models, nb_model, additive = TRUE)
+
+# variant effect ----------------------------------------------------------
+extract_variant_effect <- function(x, additive = FALSE) {  
+  samples <- posterior_samples(x, "alpha")
+  q <- samples[, "alpha"]
+  if (!additive) {
```
Comment (sbfnk) [inline R/cfr.r:171]: I find it easier to interpret q directly, i.e. 1 if no effect (and without calling it a percentage increase/decrease)

File: cfr-report.Rmd [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+  select(Type, Model = pretty_model, `ELPD difference`) %>% 
+  kbl() %>%
+  kable_styling()
+```
+*Table 1: Model comparison by difference in expected log-predictive density. Models are identified by the covariates they include.*
+
+
+```{r regression_log_coeff_table, echo = FALSE, results = 'asis'}
+to_per <- function(var) {
+  paste0(var * 100, "%")
```
Comment (sbfnk) [inline cfr-report.Rmd:126]: as above - instead of a percent change I'd just give an estimate of alpha

**Review by sbfnk — CHANGES_REQUESTED** [review]

**Comment by seabbs** [comment]
Suggest we merge this and then your branch on top of it.

**Pull request merged**